cmd/vic-machine/common/proxy_test.go

```diff
@@ -39,21 +40,29 @@ func TestProcessProxies(t *testing.T) {
 		"httpd://example.com",
 	}
 
+	gnproxies := [...]string{
+		"*",
+		"localhost, .example.com",
+		"192.168.2.5",
+	}
+
 	for _, ghttp := range gurls {
 		for _, ghttps := range gurls {
-			gproxy := Proxies{HTTPProxy: &ghttp, HTTPSProxy: &ghttps}
+			for _, uri := range gnproxies {
+				gproxy := Proxies{HTTPProxy: &ghttp, HTTPSProxy: &ghttps, NoProxy: &uri}
 
-			_, _, err := gproxy.ProcessProxies()
-			assert.NoError(t, err, "Expected %s and %s to be accepted", ghttp, ghttps)
-			assert.True(t, gproxy.IsSet, "Expected proxy to be marked as set")
+				_, _, nproxy, err := gproxy.ProcessProxies()
+				assert.NoError(t, err, "Expected %s, %s, %s and %s to be accepted", ghttp, ghttps, uri, nproxy)
```

Comment:
It seems like this will print the `--no-proxy` values twice: once from `uri` and once from `nproxy`.

Reply:
I added uri and nproxy for comparison. nproxy will trim whitespaces of each uri.